net: http: switch to zsock_ for http_client

benediktibk opened the pull request:
Using zsock_ in http_client instead of the POSIX API versions of the
functions allows the usage of http_client in combination with
CONFIG_POSIX_API.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>

rlubos commented:
@mniestroj FYI, with regards to `shutdown()`

rlubos reviewed (commented):
The changes in the library look fine, as they make the library independentof the configuration, we do this in other libraries as well (not all though yet).

I'm not conviced about the sample though, it explicitly enables `CONFIG_NET_SOCKETS_POSIX_NAMES=y` to use POSIX apis, I don't think we should change that.

benediktibk commented:
I'm uncertain if this is a desired patch, but it would resolve #38414. As the POSIX-thingie seems to be some sort of an issue (e.g. reading a bunch of comments by @pfalcon) I considered it might be easier to just avoid it internally, as it is not really necessary in this case.

benediktibk commented:
> The changes in the library look fine, as they make the library independentof the configuration, we do this in other libraries as well (not all though yet).
> 
> I'm not conviced about the sample though, it explicitly enables `CONFIG_NET_SOCKETS_POSIX_NAMES=y` to use POSIX apis, I don't think we should change that.

Okay, I will revert the changes in the sample.

benediktibk committed:
net: http: switch to zsock_ for http_client

Using zsock_ in http_client instead of the POSIX API versions of the
functions allows the usage of http_client in combination with
CONFIG_POSIX_API.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>
